cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -4120,10 +4152,8 @@ pub(crate) fn emit(
             let operand_rmi = RegMemImm::reg(operand);
             use inst_common::MachAtomicRmwOp as RmwOp;
             match op {
-                RmwOp::Xchg => {
-                    // movq %r_operand, %r_temp
-                    let i3 = Inst::mov_r_r(OperandSize::Size64, operand, temp);
-                    i3.emit(sink, info, state);
+                RmwOp::Add | RmwOp::Sub | RmwOp::Xchg => {
```

Comment:
Now that we don't use all `MachAtomicRmwOp` options in lowerings anymore, could we define a separate enum for x64 and translate into it at the rule-match step, so we don't have this problem of representable but invalid combinations?